styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -294,13 +308,46 @@ static EnvVar envVar(String name, String value) {
   @Override
   public void cleanup(WorkflowInstance workflowInstance, String executionId) {
     if (!debug.get()) {
-      LOG.info("Cleaning up {} pod: {}", workflowInstance.toKey(), executionId);
-      client.pods().withName(executionId).delete();
+      Pod pod = client.pods().withName(executionId).get();
+
+      if (pod != null) {
```

Comment:
We can probably just streamline this using `Optional` and skip that debug message.